src/utils/apiUtil.js

```diff
@@ -0,0 +1,59 @@
+const baseURL = "https://api.iextrading.com/1.0";
+
+export const api = {
+
+  getNews: symbolId =>
+    fetch(`${baseURL}/stock/${symbolId}/news/last/50`).then(
+      response => response.json()
+    ),
+
+  getOverview: symbolId =>
+    fetch(`${baseURL}/stock/${symbolId}/company`).then(
+      response => response.json()
+    ),
+
+  getChart: symbolId =>
+    fetch(`${baseURL}/stock/${symbolId}/chart/5y`).then(
+      response => response.json()
+    ),
+
+  getSymbols: () =>
+    fetch(`${baseURL}/ref-data/symbols`).then(response =>
+      response.json()
+    ),
+
+  filterChart: (symbolId, time) =>
+    fetch(
+      `${baseURL}/stock/${symbolId}/chart/${time}`
+    ).then(response => response.json()),
+
+  getMarket: () =>
+    fetch(`${baseURL}/market`).then(response =>
+      response.json()
+    ),
+
+  getKeyStats: symbolId =>
+    fetch(`${baseURL}/stock/${symbolId}/quote`).then(
+      response => response.json()
+    ),
+
+  getNews: symbolId =>
+    fetch(`${baseURL}/stock/${symbolId}/news/last/50`).then(
+      response => response.json()
+    ),
+
+  getOverview: symbolId =>
+    fetch(`${baseURL}/stock/${symbolId}/company`).then(
+      response => response.json()
+    ),
+
+  getPeers: symbolId =>
+    fetch(`${baseURL}/stock/${symbolId}/peers`).then(
+      response => response.json()
+    ),
+
+  getPrice: symbolId =>
+    fetch(`${baseURL}/stock/${symbolId}/previous`).then(
+      response => response.json()
+    )
+};
```

Comment:
I would considering adding a `request` function at the top, like this one:

```js
const request = path =>
  fetch(`${baseUrl}${path}`).then(resonpose => response.json());
```

And then expose something like:

```js
getMarket: () => request(`/market`),
getChart: symbolId => request(`/stock/${symbolId}/chart/5y`),
// etc
```

That would drastically reduce the amount of boilerplate and it would provide a common function that handles all the common logic for all the requests... Meaning that if later on we decide to have "retries" or error-handling, etc we could have put all that logic in `request` function.

Reply:
Good idea, I have refactored the api utility now